core/src/main/java/org/apache/iceberg/SchemaUpdate.java

```diff
@@ -168,6 +168,17 @@ private void internalAddColumn(
       fullName = name;
     }
 
+    // when adding structs with empty fields, Parquet fails complaining
+    // Cannot write a schema with an empty group, so fail on add column
+    // More details: https://issues.apache.org/jira/browse/SPARK-20593
```

Comment:
What's the expected behavior when the nested field (not root type) is an empty? e.g. 
```java
Types.StructType.of(Types.NestedField.optional(1, "x", Types.StructType.of()))
```
Can we add such a test?